src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,201 @@
+            var ambiguous = new List<QuickFix>();
+
+            if (compilationUnitSyntax != null)
+            {
+                usingsToAdd.UnionWith(GetUsings(root));
+
+                foreach (var unresolvedType in unresolvedTypes)
+                {
+                    candidateUsings = GetCandidateUsings(document, unresolvedType);
+
+                    //Set the symbol as an ambiguous match
+                    foreach (var candidateUsing in candidateUsings)
+                    {
+                        var unresolvedText = unresolvedType.Identifier.ValueText;
+
+                        if (!unresolvedSet.Contains(unresolvedText))
+                        {
+                            var unresolvedLocation = unresolvedType.GetLocation().GetLineSpan().StartLinePosition;
+                            ambiguous.Add(new QuickFix
+                            {
+                                Line = unresolvedLocation.Line + 1,
+                                Column = unresolvedLocation.Character + 1,
+                                FileName = fileName,
+                                Text = "`" + unresolvedText + "`" + " is ambiguous"
+                            });
+                            unresolvedSet.Add(unresolvedText);
+                        }
+                    }
+                }
+            }
+
+            return new FixUsingsResponse(compilationUnitSyntax.GetText().ToString(), ambiguous);
+        }
+
+        private static IEnumerable<SimpleNameSyntax> GetUnresolvedTypes(SemanticModel semanticModel, SyntaxNode syntaxNode)
+        {
+            return syntaxNode.DescendantNodes()
+                .OfType<SimpleNameSyntax>()
+                .Where(x => semanticModel.GetSymbolInfo(x).Symbol == null &&
+                        semanticModel.GetSymbolInfo(x).CandidateReason != CandidateReason.OverloadResolutionFailure);
+
+        }
+
+        private static bool HasLinqQuerySyntax(SemanticModel semanticModel, SyntaxNode syntaxNode)
+        {
+            return syntaxNode.DescendantNodes()
+                .Where(x => x.Kind() == SyntaxKind.QueryExpression || x.Kind() == SyntaxKind.QueryBody || x.Kind() == SyntaxKind.FromClause || x.Kind() == SyntaxKind.LetClause || x.Kind() == SyntaxKind.JoinClause || x.Kind() == SyntaxKind.JoinClause || x.Kind() == SyntaxKind.JoinIntoClause || x.Kind() == SyntaxKind.WhereClause || x.Kind() == SyntaxKind.OrderByClause || x.Kind() == SyntaxKind.AscendingOrdering || x.Kind() == SyntaxKind.DescendingOrdering || x.Kind() == SyntaxKind.SelectClause || x.Kind() == SyntaxKind.GroupClause || x.Kind() == SyntaxKind.QueryContinuation).Any();
+        }
+
+        private static HashSet<string> GetUsings(SyntaxNode root)
+        {
+            var usings = new HashSet<string>();
+            root.DescendantNodes().OfType<UsingDirectiveSyntax>()
+                .ToList().ForEach(x =>
+                        {
+                            if (!usings.Contains(x.ToString().Trim()))
```

Comment:
HashSet should automatically remove dupes. There is no need for this.

Something like this would be better I think.

```cs
var usings = root.DescendantNodes().OfType<UsingDirectiveSyntax>().Select(using => using.ToString().Trim());
return new HashSet<string>(usings);
```